### PR TITLE
[patch] refine setupdb.sh reviewed by DBA

### PR DESCRIPTION
**Pull request by csschen**
For story [#2641](https://github.ibm.com/maximoappsuite/tracker-masms/issues/2641), the maxinst script reviewed and provided by DBA [@glance](glance@ca.ibm.com), the goal is to prepare db2 for manage following the best practice.

Below is the test log.

[test.log](https://github.com/ibm-mas/ansible-devops/files/10267870/test.log)

### CODE REVIEW

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 1]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
...
+db2 update db cfg for {{ db2_dbname }} using LOGPRIMARY 100 DEFERRED
+
+### db2 update db cfg for {{ db2_dbname }} using LOGSECOND 100 DEFERRED
+db2 update db cfg for {{ db2_dbname }} using LOGSECOND 156 DEFERRED
+
+
+### db2 update db cfg for {{ db2_dbname }} using LOGFILSIZ 8192 DEFERRED
+db2 update db cfg for {{ db2_dbname }} using LOGFILSIZ 50000 DEFERRED
+
+db2 update db cfg for {{ db2_dbname }} using MIRRORLOGPATH /mnt/backup/MIRRORLOGPATH   ### Added
```
Comment (durera) [inline ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2:53]: No need to add `### Added` comments to new lines, remove the comment please.  We can see new lines in the diff tools.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 2]
```diff
@@ -44,6 +64,8 @@ db2 update db cfg for {{ db2_dbname }} using STMTHEAP AUTOMATIC DEFERRED
 db2 update db cfg for {{ db2_dbname }} using UTIL_HEAP_SZ 10000 DEFERRED
 db2 update db cfg for {{ db2_dbname }} using DATABASE_MEMORY AUTOMATIC DEFERRED
 db2 update db cfg for {{ db2_dbname }} using AUTO_STMT_STATS OFF DEFERRED
+
+###  the STMT_CONC may need to be set to off for sum customers
```
Comment (durera) [inline ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2:68]: This is good information to have, but please elaborate why & how to achieve this

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 3]
```diff
@@ -64,13 +86,24 @@ db2set DB2_FMP_COMM_HEAPSZ=65536
 db2set DB2_SKIPDELETED=ON
 db2set DB2_USE_ALTERNATE_PAGE_CLEANING=ON
 
+echo "###  set the tmp backup location   ###"
+db2set DB2_OBJECT_STORAGE_LOCAL_STAGING_PATH=/mnt/backup/staging       ### Added
```
Comment (durera) [inline ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2:90]: As previous, no need to put `### Added` comments on all new lines introduced.

**Review by durera — CHANGES_REQUESTED** [review]
How will these changes impact an existing system if this script is ran a second time?

e.g. 

- I run MAS install and the old version of this script runs
- I run a new MAs install over the top and the new version of this script runs

Will this update the settings to the new values, or will it cause problems in that db instance - e.g. because manage and/or db2 doesn't support some of these values being changed?

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 (hunk 3 above)
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed comment `### Added`

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 (hunk 1 above)
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed comment `### Added`

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 4]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
 db2 update db cfg for {{ db2_dbname }} using AUTO_MAINT ON DEFERRED
 db2 update db cfg for {{ db2_dbname }} using AUTO_TBL_MAINT ON DEFERRED
 db2 update db cfg for {{ db2_dbname }} using AUTO_RUNSTATS ON DEFERRED
-db2 update db cfg for {{ db2_dbname }} using AUTO_REORG ON DEFERRED
-db2 update db cfg for {{ db2_dbname }} using AUTO_DB_BACKUP ON DEFERRED
+
+### db2 update db cfg for {{ db2_dbname }} using AUTO_REORG ON DEFERRED
```
[the remark this answers is not in the document]
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed this commented out line.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 5]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
...
+db2 update db cfg for {{ db2_dbname }} using AUTO_REORG OFF DEFERRED
+
+
+### db2 update db cfg for {{ db2_dbname }} using AUTO_DB_BACKUP ON DEFERRED
```
[the remark this answers is not in the document]
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed this commented out line.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 6]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
...
 db2 update db cfg for {{ db2_dbname }} using CHNGPGS_THRESH 40 DEFERRED
 db2 update db cfg for {{ db2_dbname }} using DBHEAP AUTOMATIC
 db2 update db cfg for {{ db2_dbname }} using LOCKLIST AUTOMATIC DEFERRED
 db2 update db cfg for {{ db2_dbname }} using LOGBUFSZ 1024 DEFERRED
 db2 update db cfg for {{ db2_dbname }} using LOCKTIMEOUT 300 DEFERRED
-db2 update db cfg for {{ db2_dbname }} using LOGPRIMARY 20 DEFERRED
-db2 update db cfg for {{ db2_dbname }} using LOGSECOND 100 DEFERRED
-db2 update db cfg for {{ db2_dbname }} using LOGFILSIZ 8192 DEFERRED
+
+### db2 update db cfg for {{ db2_dbname }} using LOGPRIMARY 20 DEFERRED
```
[the remark this answers is not in the document]
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed this commented out line.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 7]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
...
+db2 update db cfg for {{ db2_dbname }} using LOGPRIMARY 100 DEFERRED
+
+### db2 update db cfg for {{ db2_dbname }} using LOGSECOND 100 DEFERRED
```
[the remark this answers is not in the document]
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed this commented out line.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 [hunk 8]
```diff
@@ -25,17 +25,37 @@ db2 update db cfg for {{ db2_dbname }} using dft_table_org ROW
...
+db2 update db cfg for {{ db2_dbname }} using LOGSECOND 156 DEFERRED
+
+
+### db2 update db cfg for {{ db2_dbname }} using LOGFILSIZ 8192 DEFERRED
```
[the remark this answers is not in the document]
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Removed this commented out line.

File: ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2 (hunk 2 above)
Comment (csschen) [reply ibm/mas_devops/roles/suite_db2_setup_for_manage/templates/setupdb.sh.j2]: Asked the DBA, added the comments regarding why & how to achieve this.

**Comment by csschen** [comment]
> How will these changes impact an existing system if this script is ran a second time?
> 
> e.g.
> 
> * I run MAS install and the old version of this script runs
> * I run a new MAs install over the top and the new version of this script runs
> 
> Will this update the settings to the new values, or will it cause problems in that db instance - e.g. because manage and/or db2 doesn't support some of these values being changed?

Asked the DBA, below is the answer from [@glance](glance@ca.ibm.com):

`There will be no impact, the script will simply update the parameters again. If the database is active, you will receive a message indicating a bounce will be needed in order to pickup the parameter, which is not an issue since the last step in the script is to bounce the db.`

I also tested the script on existing DB2 instance more than 2 times, the logs shown the same per time. @durera

**Comment by csschen** [comment]
Invalid PR, created a new one #576 to merge.

**Pull request closed without merging**